src/generator.rs

```diff
@@ -316,10 +283,7 @@ pub trait BaseGenerator {
         Ok(())
     }
 
-    #[cfg(all(
-        any(target_arch = "x86", target_arch = "x86_64"),
-        not(target_feature = "avx2")
-    ))]
+    #[cfg(target_feature = "sse2")]
```

Comment:
should the guard here include `not(target_feature = "avx2")` to ensure that we don't include both implementations when we have a sse2 and avx2 compatible CPU?